src/libcore/iter/mod.rs

```diff
@@ -1763,6 +1796,20 @@ impl<I> DoubleEndedIterator for Enumerate<I> where
         })
     }
 
+    #[inline]
+    #[rustc_inherit_overflow_checks]
+    fn nth_back(&mut self, n: usize) -> Option<(usize, I::Item)> {
+        self.iter.nth_back(n).map(|a| {
+            let i = self.count + n;
+            self.count = i + 1;
```

Comment:
Why is this changing the count at all?  Notice that `next_back` doesn't, and `nth_back` should only be like repeated `next_back`.  I think you can just return `(self.count + len, a)`.

Reply:
You're right; I should have actually read what `count` represented.